### DISCUSSION TITLE
How to set severity for whole package?

### DISCUSSION THREAD

**Discussion by marco-eckstein** [JosefPihrt/Roslynator#1108] [General]
Setting `dotnet_analyzer_diagnostic.category-roslynator.severity = <severity>` does not set the severity for the package _Roslynator.Formatting.Analyzers_. I don't know if it sets the severity for package _Roslynator.CodeAnalysis.Analyzers_. The 52 formatting rules [are all disabled by default](https://github.com/JosefPihrt/Roslynator/blob/main/docs/analyzers/README.md), so if I want to use them, I have to explicitly configure a lot. Issue #866 mentions the following settings, but they do not work either, or at least not for the formatting rules:
```
dotnet_analyzer_diagnostic.category-Roslynator_Core.severity = <severity>
dotnet_analyzer_diagnostic.category-Roslynator_CodeAnalysis.severity = <severity>
dotnet_analyzer_diagnostic.category-Roslynator_Formatting.severity = <severity>
```

Is there a shorthand to set the severity for all rules from _Roslynator.Formatting.Analyzers_ and _Roslynator.CodeAnalysis.Analyzers_, respectively?

**Comment by josefpihrt** [comment]
## Short answer

Read https://github.com/JosefPihrt/Roslynator/blob/main/docs/Configuration.md

## Long answer

The problem is that option `dotnet_analyzer_diagnostic.category-roslynator.severity = <severity>` can be used only for analyzers that are enabled by default (This behavior is general, not specific to Roslynator).

https://github.com/JosefPihrt/Roslynator/blob/8d2b94a0db89e85be53e869bb55337c52a3c595d/docs/Configuration.md?plain=1#L6-L7

If you use VS or VS Code extension you can use default configuration file to enable all analyzers. But you will still need to configure it because many formatting analyzers require additional option to be set (as described in `Configuration.md`).

https://github.com/JosefPihrt/Roslynator/blob/main/docs/Configuration.md#default-configuration


My general recommendation would be to copy content of https://github.com/JosefPihrt/Roslynator/blob/main/docs/Configuration.md#full-list-of-options to your `EditorConfig` file and make edits according to your needs.

**Reply by marco-eckstein** [reply]
Thanks.
